x-pack/plugins/enterprise_search/public/applications/app_search/components/document_creation/document_creation_flyout.tsx

```diff
@@ -19,6 +19,7 @@ import {
   PasteJsonText,
```

Comment:
Should the state on the flyout be reset after it's closed? It's kind of weird that the same error messages persist. And it might be nice to have the form fields reset as well.

![latest](https://user-images.githubusercontent.com/1427475/103010209-eb506680-4505-11eb-9c96-842a6db43954.gif)

Because Kea automatically re-mounts logic stores when the consuming component gets unmounted, one way to easily deal with this could be to ensure the flyout component gets unmounted when it is hidden.

```
 { isDocumentCreationOpen && <DocumentCreationFlyout /> }
```

Reply:
Actually, it looks like we already do a bunch of other resetting on `openDocumentCreation` within the logic file. You could reset `errors` in the reducer when `openDocumentCreation` is called.

We might also want to reset `textInput` and `fileInput` as well...

Again though, one way to avoid all of this resetting would be to rely on Kea remounting this component entirely when it's closed. That would let you remove all of the resetting code from the logic file.

Reply:
I'm good with resetting state. I'm not sure we can remount entirely because the logic is actually getting mounted by parent components calling `isDocumentCreationOpen`, not just on the Flyout itself.

Reply:
[just to add] the only argument I can think of for not resetting state on flyout open/close is users potentially losing data if they accidentally clicked outside the flyout without meaning to, but I don't think that's a super huge or likely use case

Reply:
https://github.com/elastic/kibana/pull/86822/commits/60b6b81b1f1b63e0df5646a272d12160ac4248c2

Reply:
Yeah. There's some other oddities with reseting. Like, if you upload a file, then navigate to the paste json screen the contents of the file you uploaded appears there. It's not like objectively a bad experience, it's just sort of an odd thing to experience.